beets/importer.py

```diff
@@ -1090,6 +1090,19 @@ def extract(self):
         archive = handler_class(util.py3_path(self.toppath), mode='r')
         try:
             archive.extractall(extract_to)
+
+            # From here:
+            # https://stackoverflow.com/questions/9813243/extract-files-from-zip-file-and-retain-mod-date
+            # fixing #4392
+
+            for f in archive.infolist():
+                # path to this extracted f-item
+                fullpath = os.path.join(extract_to, f.filename)
+                # still need to adjust the dt o/w item will have the current dt
+                date_time = time.mktime(f.date_time + (0, 0, -1))
```

Comment:
I don't yet understand the need for `+ (0, 0, -1)` here… in fact, I'm not entirely sure what type `f.date_time` is. Maybe this needs a closer look?

Reply:
I added the question and answer from the stackoverflow link